mypy CI fail in mitiq_braket

Pre-Report Checklist
--------------------

- [x] I am running the latest version of mitiq
- [x] I checked to make sure that this bug has not already been reported

Issue Description
-----------------
CI tests on master fail in `mitiq_braket/conversions` with `mypy`. 

How to Reproduce
----------------

See the [`test_validate` action](https://github.com/unitaryfund/mitiq/runs/5512635083?check_suite_focus=true). 

I cannot reproduce this locally, for me 
```
Success: no issues found in 53 source files
```


### Code Snippet

```python
Include a snippet of the mitiq code that produces the error here.
```

### Error Output

```
Run make check-types
mypy mitiq --show-error-codes
mitiq/interface/mitiq_braket/conversions.py:320: error: Argument 1 to "isclose" has incompatible type "Union[float, Basic]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:322: error: Argument 1 to "isclose" has incompatible type "Union[float, Basic]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:324: error: Argument 1 to "isclose" has incompatible type "Union[float, Basic]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:32[7](https://github.com/unitaryfund/mitiq/runs/5512635083?check_suite_focus=true#step:5:7): error: Unsupported operand types for * ("Basic" and "float")  [operator]
mitiq/interface/mitiq_braket/conversions.py:327: note: Left operand is of type "Union[float, Basic]"
mitiq/interface/mitiq_braket/conversions.py:332: error: Argument 1 to "isclose" has incompatible type "Union[float, Basic]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:335: error: Unsupported operand types for * ("Basic" and "float")  [operator]
mitiq/interface/mitiq_braket/conversions.py:335: note: Left operand is of type "Union[float, Basic]"
mitiq/interface/mitiq_braket/conversions.py:340: error: Argument 1 to "isclose" has incompatible type "Union[float, Basic]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:342: error: Argument 1 to "isclose" has incompatible type "Union[float, Basic]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:344: error: Argument 1 to "isclose" has incompatible type "Union[float, Basic]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:346: error: Argument 1 to "isclose" has incompatible type "Union[float, Basic]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:34[8](https://github.com/unitaryfund/mitiq/runs/5512635083?check_suite_focus=true#step:5:8): error: Argument 1 to "isclose" has incompatible type "Union[float, Basic]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:351: error: Unsupported operand types for * ("Basic" and "float")  [operator]
mitiq/interface/mitiq_braket/conversions.py:351: note: Left operand is of type "Union[float, Basic]"
mitiq/interface/mitiq_braket/conversions.py:354: error: Argument 1 to "abs" has incompatible type "Union[float, Basic]"; expected "SupportsAbs[float]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:400: error: Argument 1 to "abs" has incompatible type "Union[float, Basic]"; expected "SupportsAbs[float]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:404: error: Argument 1 to "abs" has incompatible type "Union[float, Basic]"; expected "SupportsAbs[float]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:408: error: Argument 1 to "isclose" has incompatible type "Union[float, Basic]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"  [arg-type]
mitiq/interface/mitiq_braket/conversions.py:4[13](https://github.com/unitaryfund/mitiq/runs/5512635083?check_suite_focus=true#step:5:13): error: Unsupported operand types for * ("Basic" and "float")  [operator]
mitiq/interface/mitiq_braket/conversions.py:413: note: Left operand is of type "Union[float, Basic]"
mitiq/interface/mitiq_braket/conversions.py:417: error: Unsupported operand types for * ("Basic" and "float")  [operator]
mitiq/interface/mitiq_braket/conversions.py:417: note: Left operand is of type "Union[float, Basic]"
mitiq/interface/mitiq_braket/conversions.py:421: error: Unsupported operand types for * ("Basic" and "float")  [operator]
mitiq/interface/mitiq_braket/conversions.py:421: note: Left operand is of type "Union[float, Basic]"
mitiq/zne/scaling/parameter.py:[15](https://github.com/unitaryfund/mitiq/runs/5512635083?check_suite_focus=true#step:5:15)1: error: Unsupported operand types for * ("Basic" and "float")  [operator]
mitiq/zne/scaling/parameter.py:151: note: Left operand is of type "Union[float, Basic]"
Found [20](https://github.com/unitaryfund/mitiq/runs/5512635083?check_suite_focus=true#step:5:20) errors in 2 files (checked 53 source files)
make: *** [Makefile:20: check-types] Error 1
Error: Process completed with exit code 2.```

Environment Context
-------------------

Use the `about()` function to summarize information on operating system, python version and dependencies.
```python
from mitiq import about
about()
```

Additional Python Environment Details (`pip freeze` or `conda list`):

```
Copy and paste the output of `pip freeze` or `conda list` here.
```

--- a/mitiq/interface/mitiq_braket/conversions.py
+++ b/mitiq/interface/mitiq_braket/conversions.py
@@ -314,9 +314,9 @@ def _translate_one_qubit_cirq_operation_to_braket_instruction(
 
     # Check common single-qubit gates.
     if isinstance(op, cirq_ops.Operation):
-        if isinstance(op.gate, cirq_ops.XPowGate):
-            exponent = op.gate.exponent
 
+        if isinstance(op.gate, cirq_ops.XPowGate):
+            exponent = cast(float, op.gate.exponent)
             if np.isclose(exponent, 1.0) or np.isclose(exponent, -1.0):
                 return [Instruction(braket_gates.X(), target)]
             elif np.isclose(exponent, 0.5):
@@ -327,16 +327,14 @@ def _translate_one_qubit_cirq_operation_to_braket_instruction(
             return [Instruction(braket_gates.Rx(exponent * np.pi), target)]
 
         elif isinstance(op.gate, cirq_ops.YPowGate):
-            exponent = op.gate.exponent
-
+            exponent = cast(float, op.gate.exponent)
             if np.isclose(exponent, 1.0) or np.isclose(exponent, -1.0):
                 return [Instruction(braket_gates.Y(), target)]
 
             return [Instruction(braket_gates.Ry(exponent * np.pi), target)]
 
         elif isinstance(op.gate, cirq_ops.ZPowGate):
-            exponent = op.gate.exponent
-
+            exponent = cast(float, op.gate.exponent)
             if np.isclose(exponent, 1.0) or np.isclose(exponent, -1.0):
                 return [Instruction(braket_gates.Z(), target)]
             elif np.isclose(exponent, 0.5):
@@ -351,7 +349,7 @@ def _translate_one_qubit_cirq_operation_to_braket_instruction(
             return [Instruction(braket_gates.Rz(exponent * np.pi), target)]
 
         elif isinstance(op.gate, cirq_ops.HPowGate) and np.isclose(
-            abs(op.gate.exponent), 1.0
+            abs(cast(float, op.gate.exponent)), 1.0
         ):
             return [Instruction(braket_gates.H(), target)]
 
@@ -397,28 +395,37 @@ def _translate_two_qubit_cirq_operation_to_braket_instruction(
 
     # Check common two-qubit gates.
     if isinstance(op.gate, cirq_ops.CNotPowGate) and np.isclose(
-        abs(op.gate.exponent), 1.0
+        abs(cast(float, op.gate.exponent)), 1.0
     ):
         return [Instruction(braket_gates.CNot(), [q1, q2])]
     elif isinstance(op.gate, cirq_ops.CZPowGate) and np.isclose(
-        abs(op.gate.exponent), 1.0
+        abs(cast(float, op.gate.exponent)), 1.0
     ):
         return [Instruction(braket_gates.CZ(), [q1, q2])]
     elif isinstance(op.gate, cirq_ops.ISwapPowGate) and np.isclose(
-        op.gate.exponent, 1.0
+        cast(float, op.gate.exponent), 1.0
     ):
         return [Instruction(braket_gates.ISwap(), [q1, q2])]
     elif isinstance(op.gate, cirq_ops.XXPowGate):
         return [
-            Instruction(braket_gates.XX(op.gate.exponent * np.pi), [q1, q2])
+            Instruction(
+                braket_gates.XX(cast(float, op.gate.exponent) * np.pi),
+                [q1, q2],
+            )
         ]
     elif isinstance(op.gate, cirq_ops.YYPowGate):
         return [
-            Instruction(braket_gates.YY(op.gate.exponent * np.pi), [q1, q2])
+            Instruction(
+                braket_gates.YY(cast(float, op.gate.exponent) * np.pi),
+                [q1, q2],
+            )
         ]
     elif isinstance(op.gate, cirq_ops.ZZPowGate):
         return [
-            Instruction(braket_gates.ZZ(op.gate.exponent * np.pi), [q1, q2])
+            Instruction(
+                braket_gates.ZZ(cast(float, op.gate.exponent) * np.pi),
+                [q1, q2],
+            )
         ]
 
     # Arbitrary two-qubit unitary decomposition.

--- a/mitiq/zne/scaling/parameter.py
+++ b/mitiq/zne/scaling/parameter.py
@@ -148,7 +148,7 @@ def scale_parameters(
             else:
                 assert isinstance(gate, EigenGate)
                 base_gate = _get_base_gate(gate)
-                param = gate.exponent * np.pi
+                param = cast(float, gate.exponent) * np.pi
                 error = rng.normal(loc=0.0, scale=np.sqrt(noise))
                 new_param = param + error
                 curr_moment.append(

